src/dex/fluid-dex/fluid-dex.ts

```diff
@@ -0,0 +1,827 @@
+import {
+  SimpleExchange,
+  getLocalDeadlineAsFriendlyPlaceholder,
+} from '../simple-exchange';
+import FluidDexPoolABI from '../../abi/fluid-dex/fluid-dex.abi.json';
+import { FluidDexConfig, Adapters, FLUID_DEX_GAS_COST } from './config';
+import { FluidDexEventPool } from './fluid-dex-pool';
+import { applyTransferFee } from '../../lib/token-transfer-fee';
+import { getDexKeysWithNetwork, getBigIntPow } from '../../utils';
+import { extractReturnAmountPosition } from '../../executor/utils';
+
+export class FluidDex extends SimpleExchange implements IDex<FluidDexData> {
+  readonly eventPools: { [id: string]: FluidDexEventPool } = {};
+
+  readonly hasConstantPriceLargeAmounts = false;
+  // TODO: set true here if protocols works only with wrapped asset
+  readonly needWrapNative = false;
+
+  readonly isFeeOnTransferSupported = false;
+
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(FluidDexConfig);
+
+  logger: Logger;
+
+  readonly pools: [FluidDexPool];
+
+  readonly iFluidDexPool: Interface;
+
+  protected adapters;
+
+  FEE_100_PERCENT = BigInt(1000000);
+
+  constructor(
+    readonly network: Network,
+    readonly dexKey: string,
+    readonly dexHelper: IDexHelper,
+  ) {
+    super(dexHelper, dexKey);
+    this.logger = dexHelper.getLogger(dexKey);
+    this.pools = FluidDexConfig['FluidDex'][network].pools;
+    this.adapters = Adapters[network] || {}; // TODO: add any additional optional params to support other fork DEXes
+    this.iFluidDexPool = new Interface(FluidDexPoolABI);
+    for (const pool of this.pools) {
+      this.eventPools[pool.id] = new FluidDexEventPool(
+        'FluidDex',
+        pool,
+        network,
+        dexHelper,
+        this.logger,
+      );
+    }
+  }
+
+  // Initialize pricing is called once in the start of
+  // pricing service. It is intended to setup the integration
+  // for pricing requests. It is optional for a DEX to
+  // implement this function
+  async initializePricing(blockNumber: number) {
+    // TODO: complete me!
+  }
+
+  // Returns the list of contract adapters (name and index)
+  // for a buy/sell. Return null if there are no adapters.
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return this.adapters[side] ? this.adapters[side] : null;
+  }
+
+  // Returns list of pool identifiers that can be used
+  // for a given swap. poolIdentifiers must be unique
+  // across DEXes. It is recommended to use
+  // ${dexKey}_${poolAddress} as a poolIdentifier
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    // TODO: complete me!
+    // return [];
+    const pool = this.getPoolByTokenPair(srcToken, destToken);
+    return pool ? [pool.id] : [];
+  }
+
+  getPoolByTokenPair(srcToken: Token, destToken: Token): FluidDexPool | null {
+    const srcAddress = srcToken.address.toLowerCase();
```

Comment:
same as getPoolByTokenPairAddress? let's just call getPoolByTokenPairAddress passing in address?